client/jetpack-cloud/sections/partner-portal/credit-card-fields/index.tsx

```diff
@@ -0,0 +1,109 @@
+import { FormStatus, useFormStatus } from '@automattic/composite-checkout';
+import { Field } from '@automattic/wpcom-checkout';
+import { StripeElementChangeEvent, StripeElementStyle } from '@stripe/stripe-js';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useI18n } from '@wordpress/react-i18n';
+import classnames from 'classnames';
+import { useState } from 'react';
+import { useDispatch as useReduxDispatch } from 'react-redux';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import CreditCardElementField from './credit-card-element-field';
+import CreditCardLoading from './credit-card-loading';
+import SetAsPrimaryPaymentMethod from './set-as-primary-payment-method';
+
+import './style.scss';
+
+export default function CreditCardFields() {
+	const { __ } = useI18n();
+	const [ isStripeFullyLoaded, setIsStripeFullyLoaded ] = useState( false );
+	const fields = useSelect( ( select ) => select( 'credit-card' ).getFields() );
+	const useAsPrimaryPaymentMethod = useSelect( ( select ) =>
+		select( 'credit-card' ).useAsPrimaryPaymentMethod()
+	);
+	const getField = ( key: string | number ) => fields[ key ] || {};
+	const getFieldValue = ( key: string | number ) => getField( key ).value ?? '';
+	const getErrorMessagesForField = ( key: string | number ) => {
+		const managedValue = getField( key );
+		return managedValue.errors ?? [];
+	};
+	const {
+		setFieldValue,
+		setCardDataError,
+		setCardDataComplete,
+		setUseAsPrimaryPaymentMethod,
+	} = useDispatch( 'credit-card' );
+	const reduxDispatch = useReduxDispatch();
+
+	const cardholderName = getField( 'cardholderName' );
+	const cardholderNameErrorMessages = getErrorMessagesForField( 'cardholderName' ) || [];
+	const cardholderNameErrorMessage = cardholderNameErrorMessages.length
+		? cardholderNameErrorMessages[ 0 ]
+		: null;
+
+	const handleStripeFieldChange = ( input: StripeElementChangeEvent ) => {
+		setCardDataComplete( input.elementType, input.complete );
+
+		if ( input.error && input.error.message ) {
+			reduxDispatch(
+				recordTracksEvent( 'calypso_partner_portal_stripe_field_invalid_error', {
+					error_type: 'Stripe field error',
+					error_field: input.elementType,
+					error_message: input.error.message,
+				} )
+			);
+			setCardDataError( input.elementType, input.error.message );
+			return;
+		}
+		setCardDataError( input.elementType, null );
+	};
+
+	const stripeElementStyle: StripeElementStyle = {
+		base: {
+			color: '#2c3338',
+		},
+		invalid: {
+			color: '#D63638',
+		},
+	};
+
+	const { formStatus } = useFormStatus();
+	const isDisabled = formStatus !== FormStatus.READY;
+
+	return (
+		<>
+			{ ! isStripeFullyLoaded && <CreditCardLoading /> }
+
+			<div
+				className={ classnames( {
+					'credit-card-fields': true,
+					'credit-card-fields--is-loaded': isStripeFullyLoaded,
```

Comment:
I see that you're controlling the appearance of the `div` element via CSS with `display: none | display: block`, but you can achieve the same result in a more readable way if you prefix that entire HTML section with `{ isStripeFullyLoaded && ( /* your HTML goes here... */ ) }`. Otherwise, readers need to go to your CSS to realize that.

Reply:
We’re using CSS because the credit card element itself has some initialization work to do, and we consider Stripe as loaded only when [this element tells us about it](https://github.com/Automattic/wp-calypso/blob/add/licensing-portal-credit-card-form/client/jetpack-cloud/sections/partner-portal/credit-card-fields/credit-card-element-field.tsx#L41). If we don’t render the element, we won't know when the initialization will be finished.